client/src/components/TestSuite/TestSession.tsx

```diff
@@ -144,13 +166,39 @@ const TestSessionComponent: FC<TestSessionComponentProps> = ({ testSession, prev
   function createTestRun(runnableType: RunnableType, runnableId: string, inputs: TestInput[]) {
     postTestRun(id, runnableType, runnableId, inputs)
       .then((testRun: TestRun) => {
-        saveTestRunResults(testRun);
+        setTestRun(testRun);
+        setShowProgressBar(true);
+        pollTestRunResults(testRun);
       })
       .catch((e) => {
         console.log(e);
       });
   }
 
+  const completedTestCount = testRun?.results?.filter((result) => result.test_id)?.length || 0;
+
+  function testRunNeedsProgressBar(testRun: TestRun | null) {
+    return testRun?.status === 'running';
+  }
+
+  function testRunProgressBar() {
+    const duration = testRunNeedsProgressBar(testRun) ? null : 2000;
+    return (
+      <Snackbar
```

Comment:
I think the snackbar should be in the TestRunProgressBar component because I think of those as one entity, and it will make this file cleaner.

Reply:
I will address this in #24